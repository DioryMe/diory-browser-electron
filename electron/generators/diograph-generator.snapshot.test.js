const { join } = require('path')
const uuid = require('uuid')
const { generateDiograph } = require('./diograph-generator')

jest.mock('uuid')

describe('diograph-generator', () => {
  let act

  describe('generateDiograph', () => {
    let folderPath
    let paths
    let folderDiory

    it('generates diograph', async () => {
      ;[...Array(10).keys()].forEach(id => {
        uuid.mockReturnValueOnce(`uuid-${id}`)
      })

      const { diograph } = await generateDiograph(join(__dirname, '../readers/example-folder'))

      expect(diograph).toMatchSnapshot()
    })
  })
})
