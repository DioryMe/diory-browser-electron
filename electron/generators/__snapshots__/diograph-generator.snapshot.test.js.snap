// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diograph-generator generateDiograph generates diograph from example folder 1`] = `
Object {
  "uuid-0": Object {
    "created": "created-timestamp",
    "date": "2019-07-09T00:01:16.000Z",
    "id": "uuid-0",
    "image": "/electron/readers/example-folder/image-with-24-hour.jpg",
    "latitude": 68.27818055555555,
    "longitude": 14.302322222222223,
    "modified": "modified-timestamp",
  },
  "uuid-1": Object {
    "created": "created-timestamp",
    "date": "2008-11-01T21:15:11.000Z",
    "id": "uuid-1",
    "image": "/electron/readers/example-folder/other-image.jpg",
    "latitude": 43.464455,
    "longitude": 11.881478333333334,
    "modified": "modified-timestamp",
  },
  "uuid-2": Object {
    "created": "2019-11-25T19:58:24.734Z",
    "date": "2019-11-25T19:58:24.000Z",
    "id": "uuid-2",
    "image": "/electron/readers/example-folder/some-image.jpg",
    "modified": "modified-timestamp",
  },
  "uuid-3": Object {
    "created": "created-timestamp",
    "id": "uuid-3",
    "modified": "modified-timestamp",
    "text": "some-text",
  },
  "uuid-4": Object {
    "created": "2019-10-27T14:40:43.879Z",
    "date": "2019-10-27T14:40:43.000Z",
    "id": "uuid-4",
    "image": "/electron/readers/example-folder/some-folder/some-folder-image.jpg",
    "latitude": 61.49578611111111,
    "longitude": 23.76756111111111,
    "modified": "modified-timestamp",
  },
  "uuid-5": Object {
    "created": "created-timestamp",
    "date": "2019-10-27T14:40:43.000Z",
    "id": "uuid-5",
    "image": "/electron/readers/example-folder/some-folder/some-folder-image.jpg",
    "latitude": 61.49578611111111,
    "links": Object {
      "some-folder-image.jpg": Object {
        "id": "uuid-4",
      },
    },
    "longitude": 23.76756111111111,
    "modified": "modified-timestamp",
    "text": "some-folder",
  },
  "uuid-6": Object {
    "created": "created-timestamp",
    "date": "2017-01-07T07:58:53.500Z",
    "id": "uuid-6",
    "image": "/electron/readers/example-folder/some-folder/some-folder-image.jpg",
    "latitude": 57.74614055555556,
    "links": Object {
      "image-with-24-hour.jpg": Object {
        "id": "uuid-0",
      },
      "other-image.jpg": Object {
        "id": "uuid-1",
      },
      "some-folder": Object {
        "id": "uuid-5",
      },
      "some-image.jpg": Object {
        "id": "uuid-2",
      },
      "some-text.txt": Object {
        "id": "uuid-3",
      },
    },
    "longitude": 16.65045388888889,
    "modified": "modified-timestamp",
    "text": "example-folder",
  },
}
`;
